includes/backends/class-omise-backend-touch-n-go.php

```diff
@@ -0,0 +1,37 @@
+<?php
+/**
+ *
+ * @method public initiate
+ * @method public get_provider
+ */
+class Omise_Backend_TouchNGo extends Omise_Backend {
+	/**
+	 * @var array  of known providers.
+	 */
+	protected static $providers = array();
+
+	public function initiate() {
+		self::$providers = array();
+	}
+
+	/**
+	 * 
+	 * @return string of touch n go provider
+	 */
+	public function get_provider() {
+        $tng = $this->capabilities()->getTouchNGoBackends();
+		$first_value = reset($tng);
+
+		// Preventing the following error:
+		// Uncaught TypeError: property_exists(): Argument #1 must be of type object|string, bool given
+		$typeofFirstValue = gettype($first_value);
+		$isObjectOrString = 'object' === $typeofFirstValue || 'string' === $typeofFirstValue;
+
+        $provider = null;
+		if ($isObjectOrString && property_exists($first_value, 'provider')) {
+            $provider = $first_value->provider;
+		}
+
+        return $provider;
+	}
```

Comment:
I think the indentation for this file is a bit strange 🤔

Reply:
Yea, it's strange. It look normal on VS Code 😅. Fixed [b3e2088](https://github.com/omise/omise-woocommerce/pull/287/commits/b3e20886d48e3f77e411c6e477865dcaaff47a0c)